daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/processors/unparsers/ExpressionEvaluatingUnparsers.scala

```diff
@@ -103,13 +102,22 @@ class NewVariableInstanceStartUnparser(override val context: VariableRuntimeData
   override lazy val childProcessors = Vector()
 
   override def unparse(state: UState) = {
-    state.variableMap.newVariableInstance(context)
+    val v = state.variableMap.newVariableInstance(context)
+
     if (context.maybeDefaultValueExpr.isDefined) {
-      val maybeVar = state.variableMap.find(context.globalQName)
-      Assert.invariant(maybeVar.isDefined)
-      maybeVar.get.setState(VariableInProcess)
-      val suspendableExpression = new NewVariableInstanceSuspendableExpression(context.maybeDefaultValueExpr.get, context)
-      suspendableExpression.run(state)
+      val dve = context.maybeDefaultValueExpr.get
+      dve match {
+        case constExpr: ConstantExpression[_] => // Do nothing, value has already been computed at schema compile time
+        case _ => {
+          v.setState(VariableInProcess)
+          val suspendableExpression = new NewVariableInstanceDefaultValueSuspendableExpression(dve, context)
```

Comment:
Can we pass in the variable instance instance the NewVariableInstanceDefaultValue.... so that we don't have to do a variableMap.find in processExpressionResult?